p2p/src/authenticated/discovery/actors/peer/actor.rs

```diff
@@ -42,7 +42,7 @@ pub struct Actor<E: Spawner + Clock + ReasonablyRealtime + Metrics, C: PublicKey
     rate_limited: Family<metrics::Message, Counter>,
 
     // When reservation goes out-of-scope, the tracker will be notified.
-    reservation: Reservation<E, C>,
+    _reservation: Reservation<E, C>,
```

Comment:
I think we can avoid even passing this into `Peer`. We can just hold onto it in the spawner task we create (dropping it when we send the disconnect to router).